types/draft.ts

```diff
@@ -0,0 +1,32 @@
+import { z } from 'zod'
+const Draft = z.object({
+	title: z.string().min(1).max(100),
+	summary: z.string().min(10).max(255),
+	description: z.string().min(10).max(1000),
+	type: z.enum(['topic', 'project']),
+	id: z.number().int().positive(),
+	created: z.number().transform(val => new Date(val)),
+	updated: z.number().transform(val => new Date(val)).nullable(),
+})
+
+type Draft = z.infer<typeof Draft>
+
+const PendingDraft = z.object({
+	title: z.string().min(1).max(100),
+	summary: z.string().min(10).max(255),
+	description: z.string().min(1).max(1000),
+	type: z.enum(['topic', 'project']),
+})
+
+type PendingDraft = z.infer<typeof PendingDraft>
+
+const DraftUpdate = z.object({
```

Comment:
optional refactor: look at https://github.com/colinhacks/zod?tab=readme-ov-file#partial

I've been meaning to come back for this on proposals. I think a `DraftUpdate` is just a `PendingDraft.partial()`